### PR TITLE
improve useForm's trigger type

### PR DESCRIPTION
**Pull request by kotarella1110**

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -537,7 +538,11 @@ export function useForm<
         });
       }
     } else {
-      setInternalValues(name, value, options);
+      setInternalValues(
+        (name as any) as FieldPath<TFieldValues>,
```
Comment (bluebill1049) [inline src/useForm.ts:542]: `name as FieldPath<TFieldValues>` enough?

Comment (bluebill1049) [reply src/useForm.ts:542]: we are trying to avoid `as any` in the codebase in v7 😀

Comment (kotarella1110) [reply src/useForm.ts:542]: Not enough.

スクリーンショット 2021-03-02 16 10 25

After improving setValue type, I think I can remove `as`.

Comment (bluebill1049) [reply src/useForm.ts:542]: kk, let's get it.

**Review by bluebill1049 — APPROVED** [review]
👍

**Comment by kotarella1110** [comment]
@bluebill1049 I created the branch `improve-use-Form-methods-types`.
You can do your final review on this branch.

**Pull request merged**